rfcs/0001-multi-tenancy/README.md

```diff
@@ -0,0 +1,166 @@
+With soft multi-tenancy, the platform admins use Kubernetes constructs such as namespaces, accounts,
+roles and role bindings to create a logical separation between tenants.
+
+When Flux deploys workloads from a repository belonging to a tenant, it uses the Kubernetes account assigned to that
+tenant to perform the git-to-cluster reconciliation. By leveraging Kubernetes RBAC, Flux can ensure
+that the operations performed by tenants are restricted to their namespaces.
+
+Note that with this model, tenants share cluster-wide resources such as
+`ClusterRoles`, `CustomResourceDefinitions`, `IngressClasses`, `StorageClasses`,
+and they cannot create or alter these resources.
+If a tenant adds o cluster-scoped resource definition to their repository,
+Flux will fail the git-to-cluster reconciliation due to Kubernetes RBAC restrictions.
+
+To restrict the reconciliation of tenant's sources, a Kubernetes service account name can be specified 
+in Flux `Kustomizations` and `HelmReleases` under `.spec.serviceAccountName`. Please consult the Flux  
+documentation for more details:
+
+- [Kustomization API: Role-based access control](https://fluxcd.io/docs/components/kustomize/kustomization/#role-based-access-control)
+- [HelmRelease API: Role-based access control](https://fluxcd.io/docs/components/helm/helmreleases/#role-based-access-control)
+- [Flux multi-tenancy example repository](https://github.com/fluxcd/flux2-multi-tenancy)
+
+Note that with soft multi-tenancy, true tenant isolation requires security measures beyond Kubernetes RBAC.
+Please refer to the Kubernetes [security considerations documentation](https://kubernetes.io/blog/2021/04/15/three-tenancy-models-for-kubernetes/#security-considerations)
+for more details on how to harden shared clusters.
+
+##### Caveats
+
+As of v0.23.0, Flux does not enforce a service account to be specified on Flux `Kustomizations` and `HelmReleases`.
+When a service account is not specified, Flux defaults to cluster-admin.
+In order to enforce the tenant isolation, an admission controller such as Kyverno or OPA Gatekeeper must be used
+to make the `.spec.serviceAccountName` a required field for the Flux custom resources created by tenants.
+
+We provide an [example](https://github.com/fluxcd/flux2-multi-tenancy/blob/main/infrastructure/kyverno-policies/flux-multi-tenancy.yaml) 
+for enforcing service accounts using a Kyverno cluster policy.
+
+As of v0.23.0, Flux allows for `Kustomizations` and `HelmReleases` to reference sources
+(`GitRepositories`, `HelmRepositories` and `Buckets`) across namespaces.
+In order to prevent tenants from accessing each other sources, an admission controller such as Kyverno or OPA Gatekeeper
+must be used to block cross-namespace references.
+
+We provide an [example](https://github.com/fluxcd/flux2-multi-tenancy/blob/main/infrastructure/kyverno-policies/flux-multi-tenancy.yaml)
+for blocking source cross-namespace references using a Kyverno cluster policy.
+
+#### Hard Multi-Tenancy
```

Comment:
Is this scope of this proposal covering hard multi-tenancy? I see no further mention of how this proposal will affect this scenario, so is this beyond the scope of the RFCs

Reply:
This proposal is for defining what Flux currently supports. RFC 2 and 3 cover soft multi-tenancy improvements.